### PR TITLE
add tests workflow file

### PR DESCRIPTION
**Pull request by MaHaWo**
- add a workflow file for unit tests (and others later on)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+    unit-tests: 
+      runs-on: ubuntu-latest
+
+      steps: 
+        - name: Checkout code
+          uses: actions/checkout@v4
+
+        - name: Set up environment
```
Comment (lkeegan) [inline .github/workflows/ci.yml:18]: pnpm is a faster replacement for npm, see e.g. this workflow: https://github.com/spatial-model-editor/spatial-model-editor.github.io/blob/3d93eac74e8f2b0e1b29429cb5c849602bb19093/.github/workflows/deploy.yml#L11-L20

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        - name: Set up environment 
+          uses: actions/setup-node@v4
+          with: 
+            node-version: '20'
+
+        - name: Install dependencies
+          run: npm install
+
+        - name: Run tests
+          run: npx vitest
```
Comment (lkeegan) [inline .github/workflows/ci.yml:27]: this is already a command in package.json, so we can just do `npm test` (or `pnpm test`)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MaHaWo) [reply .github/workflows/ci.yml]: :+1:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MaHaWo) [reply .github/workflows/ci.yml]: :+1:

**Pull request merged**